Sources/StreamChat/WebSocketClient/Events/ReactionEvents_Tests.swift

```diff
@@ -52,3 +53,128 @@ class ReactionEvents_Tests: XCTestCase {
+    }
+
+    func test_ReactionNewEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "ReactionNew")
+        let event = try eventDecoder.decode(from: json) as? ReactionNewEvent
+        
+        // For message to be received, we need to have channel:
+        try client.databaseContainer.createChannel(
+            cid: .init(type: .messaging, id: "general"),
+            withMessages: true,
+            withQuery: false
+        )
+        
+        try client.databaseContainer.createMessage(
+            id: "0e042a9c-d648-4a28-8ed6-dbdb2b7b4779", cid: .init(type: .messaging, id: "general")
+        )
+        
+        XCTAssertTrue(
+            client.databaseContainer.viewContext.message(id: "0e042a9c-d648-4a28-8ed6-dbdb2b7b4779")?.reactions.isEmpty ?? false
+        )
+        
+        let unwrappedEvent = try XCTUnwrap(event)
+        client.eventNotificationCenter.process(unwrappedEvent)
+
+        AssertAsync {
+            Assert.willBeFalse(
+                self.client.databaseContainer.viewContext.message(
+                    id: "0e042a9c-d648-4a28-8ed6-dbdb2b7b4779"
+                )?.reactions.isEmpty ?? true
+            )
+        }
+    }
+    
+    func test_ReactionUpdatedEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "ReactionUpdated")
+        let event = try eventDecoder.decode(from: json) as? ReactionUpdatedEvent
+        
+        let newReactionJSON = XCTestCase.mockData(fromFile: "ReactionNew")
+        let newReactionEvent = try eventDecoder.decode(from: newReactionJSON) as? ReactionNewEvent
+        let newReactionPayload = try XCTUnwrap((newReactionEvent?.payload as? EventPayload<NoExtraData>)?.reaction)
+        
+        // For message to be received, we need to have channel:
+        try client.databaseContainer.createChannel(
+            cid: .init(type: .messaging, id: "general"),
+            withMessages: true,
+            withQuery: false
+        )
+    
+        let lastUpdateMessageTime = "2020-06-20 17:09:56 +0000"
+        
+        try client.databaseContainer.createMessage(
+            id: "0e042a9c-d648-4a28-8ed6-dbdb2b7b4779",
+            latestReactions: [newReactionPayload],
+            type: .regular
+        )
+        
+        XCTAssertEqual(
+            client.databaseContainer.viewContext.message(
+                id: "0e042a9c-d648-4a28-8ed6-dbdb2b7b4779"
+            )?.reactions.first?.updatedAt.description,
+            lastUpdateMessageTime
+        )
+        
+        let unwrappedUpdate = try XCTUnwrap(event)
+        client.eventNotificationCenter.process(unwrappedUpdate)
+        
+        AssertAsync {
+            Assert.willBeEqual(
+                self.client.databaseContainer.viewContext.message(
+                    id: "0e042a9c-d648-4a28-8ed6-dbdb2b7b4779"
+                )?.reactions.first?.updatedAt.description,
+                "2020-07-20 17:09:56 +0000"
```

Comment:
We have `String.toDate()` test helper for converting stream backend strings to `Date`

Reply:
I guess I will leave this to part of polishing...